plugins/outputs/bigquery/bigquery_test.go

```diff
@@ -0,0 +1,96 @@
+package bigquery
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"net/http/httputil"
+	"os"
+	"strings"
+	"testing"
+
+	"cloud.google.com/go/bigquery"
+	"github.com/influxdata/telegraf/testutil"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/api/option"
+)
+
+const (
+	successfulResponse = "{\"kind\": \"bigquery#tableDataInsertAllResponse\"}"
+)
+
+var testingHost string
+
+func TestMain(t *testing.M) {
+	srv := localBigQueryServer(t)
+
+	testingHost = strings.ReplaceAll(srv.URL, "http://", "")
+
+	defer srv.Close()
+
+	os.Exit(t.Run())
+}
```

Comment:
I don't get the intention behind this test... If you only want to test if you can setup a local test-server, I think you don't need to do it.

Reply:
I run the tests through TestMain. I use TestMain in order to spin up the server that produces a fake BigQuery Server. Then through T.Run() the tests `TestWrite` and  `TestConnect` will run using the fake BigQuery Server.

Reply:
This is an unusual pattern for telegraf. :-) Is there any reason not to spin up a server in every test (that's what everybody else in telegraf does)?

Reply:
Changed to spinning up per test.